dipy/workflows/base.py

```diff
@@ -0,0 +1,131 @@
+        conflict_handler : str
+            String indicating how to handle conflicts
+        add_help : bool
+            Add a -h/-help option
+        """
+
+        iap = IntrospectiveArgumentParser
+        super(iap, self).__init__(prog, usage, description, epilog, version,
+                                  parents, formatter_class, prefix_chars,
+                                  fromfile_prefix_chars, argument_default,
+                                  conflict_handler, add_help)
+
+        self.doc = None
+
+    def add_workflow(self, workflow):
+        specs = inspect.getargspec(workflow)
+        doc = inspect.getdoc(workflow)
+        self.doc = NumpyDocString(doc)['Parameters']
+
+        args = specs.args
+        defaults = specs.defaults
+
+        len_args = len(args)
+        len_defaults = len(defaults)
+
+        # Arguments with no defaults (Positional)
+        cnt = 0
+        for i in range(len_args - len_defaults):
+            typestr = self.doc[i][1]
+            dtype = self._select_dtype(typestr)
+            help_msg = ''.join(self.doc[i][2])
+            if dtype is bool:
+                self.add_argument(args[i], choices=[0, 1], type=int,
+                                  action='store', metavar=dtype.__name__,
+                                  help=help_msg)
+            else:
+                self.add_argument(args[i], action='store',
+                                  type=dtype, metavar=dtype.__name__,
+                                  help=help_msg)
+            cnt += 1
+
+        # Arguments with defaults (Optional)
+        for i in range(cnt, len_args):
+            typestr = self.doc[i][1]
+            dtype = self._select_dtype(typestr)
+            help_msg = ' '.join(self.doc[i][2])
+
+            if dtype is bool:
+                self.add_argument('--' + args[i], choices=[0, 1], type=int,
+                                  action='store', metavar=dtype.__name__,
+                                  help=help_msg)
+            else:
+                self.add_argument('--' + args[i], action='store',
+                                  type=dtype, metavar=dtype.__name__,
+                                  help=help_msg)
+
+    def _select_dtype(self, text):
+        text = text.lower()
+        if 'str' in text:
+            return str
+        if 'int' in text:
+            return int
+        if 'float' in text:
+            return float
+        if 'bool' in text:
+            return bool
+
+    def get_flow_args(self, args=None, namespace=None):
+        ns_args = self.parse_args(args, namespace)
+        dct = vars(ns_args)
+
+        return dict((k, v) for k, v in dct.items() if v is not None)
+
+    def update_argument(self, *args, **kargs):
+
+        self.add_argument(*args, **kargs)
+
+    def show_argument(self, dest):
+
+        for act in self._actions[1:]:
+            if act.dest == dest:
+                print(act)
+
+    def add_epilogue(self):
+        # with citations
```

Comment:
Hola! Can you include the Frontiers Dipy paper as default epilogue here?
Garyfallidis, E., M. Brett, B. Amirbekian, A. Rokem, S. Van Der Walt, M. Descoteaux, and I. Nimmo-Smith. Dipy, a library for the analysis of diffusion MRI data. Frontiers in Neuroinformatics, 1-18, 2014.
So, if no papers are added in the epilogue or description, people should be at least citing Dipy.